### ISSUE TITLE
In dataset completion has more than 4096 tokens model hugging face meta-llama/Llama-2-7b-chat-hf

### ISSUE THREAD

**Issue by yogesh1273** [meta-llama/llama-recipes#268]
I am using meta-llama/Llama-2-7b-chat-hf model for code generation. My inference time from the trained model is about 4-5 minutes by using pipelines and with model.generate() 10-15 mins. And everything is saved locally.
My dataset has 125 rows it's a jsonl file and each row has prompt and completion, but in the completion I have more than 4096 tokens.
I have 8 GPUs of 32gb TESLA V100. While inference sometimes it just gives random answer after completion or copy the last line multiple times.
This is my training code:
`import torch
from transformers import (
AutoTokenizer,
AutoModelForCausalLM,
BitsAndBytesConfig,
HfArgumentParser,
TrainingArguments,
pipeline,
logging,
)
from peft import LoraConfig, PeftModel
from trl import SFTTrainer
from datasets import load_dataset

model_name = "NousResearch/Llama-2-7b-chat-hf"

dataset_name = "train.jsonl"

new_model = "llama-2-7b-finetuned"

output_dir = "./results"

num_train_epochs = 8

bnb_config = BitsAndBytesConfig(
load_in_4bit=True,
bnb_4bit_quant_type="nf4",
bnb_4bit_compute_dtype=torch.float16,
bnb_4bit_use_double_quant=False,

model = AutoModelForCausalLM.from_pretrained(
model_name,
quantization_config=bnb_config,
# use the gpu
device_map= "auto"
)

model.config.use_cache = False

tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True, use_fast=False)
tokenizer.pad_token = tokenizer.eos_token
tokenizer.padding_side = "right"

dataset = load_dataset("json", data_files="train.jsonl")
dataset = dataset["train"]

peft_config = LoraConfig(
lora_alpha=16,
lora_dropout=0.1,
r=64,
bias="none",
task_type="CAUSAL_LM",
)

training_arguments = TrainingArguments(
output_dir=output_dir,
num_train_epochs=num_train_epochs, # uses the number of epochs earlier
per_device_train_batch_size=2, # 4 seems reasonable
gradient_accumulation_steps=1, # 2 is fine, as we're a small batch
optim="paged_adamw_32bit", # default optimizer
save_steps=0, # we're not gonna save
logging_steps=10, # same value as used by Meta
learning_rate=2e-4, # standard learning rate
weight_decay=0.001, # standard weight decay 0.001
fp16=False, # set to true for A100
bf16=False, # set to true for A100
max_grad_norm=0.3, # standard setting
max_steps=-1, # needs to be -1, otherwise overrides epochs
warmup_ratio=0.03, # standard warmup ratio
group_by_length=True, # speeds up the training
lr_scheduler_type="cosine", # constant seems better than cosine # TODO: why not constant?
report_to="tensorboard"
)

trainer = SFTTrainer(
model=model,
train_dataset=dataset,
peft_config=peft_config, # use our lora peft config
dataset_text_field="text",
max_seq_length=4096, # no max sequence length
tokenizer=tokenizer, # use the llama tokenizer
args=training_arguments, # use the training arguments
packing=False, # don't need packing
)

trainer.train()

trainer.model.save_pretrained(new_model)
trainer.tokenizer.save_pretrained(new_model)

del model
del trainer

import gc
gc.collect()

base_model = AutoModelForCausalLM.from_pretrained(
model_name,
return_dict=True,
torch_dtype=torch.float16,
device_map="auto",
)
model = PeftModel.from_pretrained(base_model, new_model)
model = model.merge_and_unload()

model.save_pretrained("the-final4")
tokenizer.save_pretrained("the-final4")`

and this is my Inference code:
`# Fine-tuned model name
model_name = "./the-final4"

import torch
from transformers import (
AutoTokenizer,
AutoModelForCausalLM,
BitsAndBytesConfig,
pipeline,
logging,
)

bnb_config = BitsAndBytesConfig(
load_in_4bit=True,
bnb_4bit_quant_type="nf4",
bnb_4bit_compute_dtype=torch.float16,
bnb_4bit_use_double_quant=False,
)

model = AutoModelForCausalLM.from_pretrained(
model_name,
quantization_config=bnb_config,
device_map= "auto"
)

tokenizer = AutoTokenizer.from_pretrained(model_name)#, trust_remote_code=True, use_fast=False)

model.use_cache = True

model.eval()
print("hello")
prompt = "Implement a code snippet that creates a window handle message for the textbox widget."
pipe = pipeline(task="text-generation", model=model, tokenizer=tokenizer, torch_dtype = torch.float16)
result = pipe(f " [INST] {prompt} [/INST] ", do_sample = True, top_p = 0.9, temperature = 0.6 ,num_return_sequences = 1, eos_token_id = tokenizer.eos_token_id,max_length = 4096)
print(result[0]['generated_text']) `

**Comment by yogesh1273** [comment]
yes, that's right. But the answer which it gives is not correct also. And i am doing this for code generation and some of my codes are more than 4096 tokens
